diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +124,169 @@ class LineLength : Rule("line-length") {
+        var leftOffset = if (leftInitOffset < 0) {
+            positionByOffset(wrongNode.firstChildNode.startOffset).second
+        } else leftInitOffset
+        val binList = mutableListOf<ASTNode>()
+        if (isParenthesized)
+            dfsForProperty(wrongNode, binList)
+        else
+            searchBinaryExpression(wrongNode, binList)
+        var binaryText = ""
+        binList.forEachIndexed { index, astNode ->
+            if (index == 0) {
+                binaryText += astNode.treeParent.prevSibling { it.elementType == LPAR }?.text ?: ""
+                binaryText += astNode.prevSibling { it.elementType == WHITE_SPACE }?.text ?: ""
+                binaryText += astNode.text
+                binaryText += astNode.nextSibling { it.elementType == WHITE_SPACE }?.text ?: ""
+            } else {
+                binaryText += astNode.treeParent.prevSibling { it.elementType == LPAR }?.text ?: ""
+                binaryText += astNode.prevSibling { it.elementType == WHITE_SPACE }?.text ?: ""
+                binaryText += astNode.treeParent.findChildByType(OPERATION_REFERENCE)!!.text
+                binaryText += astNode.text
+                binaryText += astNode.nextSibling { it.elementType == WHITE_SPACE }?.text ?: ""
+                binaryText += astNode.treeParent.nextSibling { it.elementType == RPAR }?.text ?: ""
+                if (leftOffset + binaryText.length > configuration.lineLength) {
+                    val commonParent = astNode.parent({ it in binList[index - 1].parents() })!!
+                    val nextNode = commonParent.findChildByType(OPERATION_REFERENCE)!!.treeNext
+                    if (!nextNode.text.contains("\n"))
+                        commonParent.appendNewlineMergingWhiteSpace(nextNode, nextNode)
+                    leftOffset = 0
+                    binaryText = astNode.text
+                    return
+                }
+            }
+        }
+    }
+
+    /**
+     * This method stored all the nodes that have BINARY_EXPRESSION or PREFIX_EXPRESSION element type.
+     * This method uses recursion to store binary node in the order in which they are located
+     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
+     *@param node node in which to search
+     *@param binList mutable list of ASTNode to store nodes
+     */
+    private fun searchBinaryExpression(node: ASTNode, binList: MutableList<ASTNode>) {
+        if (node.hasChildOfType(BINARY_EXPRESSION) || node.hasChildOfType(PARENTHESIZED)) {
+            node.getChildren(null)
+                    .filter { it.elementType == BINARY_EXPRESSION || it.elementType == PARENTHESIZED }
+                    .forEach {
+                        searchBinaryExpression(it, binList)
+                    }
+        } else {
+            binList.add(node)
+            binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
```

Comment:
What about `POSTFIX_EXPRESSIONS` like references with `!!`? Please check how they are handled and add test case.